extensions/agroal/deployment/src/test/java/io/quarkus/agroal/test/AgroalMetricsOnlyTestCase.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkus.agroal.test;
+
+import java.sql.SQLException;
+
+import jakarta.inject.Inject;
+
+import org.eclipse.microprofile.metrics.*;
+import org.eclipse.microprofile.metrics.annotation.RegistryType;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+/**
+ * Test to verify that Agroal metrics are exposed when only quarkus.datasource.metrics.enabled=true is set
+ */
+public class AgroalMetricsOnlyTestCase {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .withConfigurationResource("application-metrics-only-enabled.properties");
```

Comment:
It would be nice to use `.overrideConfigKey( )` on some basic configuration e.g. `base.properties`, so to avoid an explosion of properties files with all permutations.

It also makes it better maintainable, as then people understand which property exactly is this particular test focusing on.